appscale/tools/agents/azure_agent.py

```diff
@@ -1193,3 +1214,92 @@ def does_resource_group_exist(self, resource_group_name, resource_client):
     if resource_group_name in resource_group_names:
       return True
     return False
+
+  def create_image(self, instance_id, image_name, parameters, dest_container):
+    """ Creates an Azure virtual machine using the network interface created.
+   Args:
+     instance_id: The instance id of the VM running.
+     image_name: The prefix to give the VHD we create.
+     parameters: A dict, containing all the parameters necessary to
+       authenticate this user with Azure.
+     dest_container: The container to store the VHD we create in. Stored in
+       the format:
+      {resource_group}/system/Microsoft.Compute/Images/{dest_container}
+   Raises:
+     IndexError, AttributeError if VirtualMachineCaptureResult is not
+      formatted as expected.
+   """
+    credentials = self.open_connection(parameters)
+    resource_group = parameters[self.PARAM_RESOURCE_GROUP]
+    subscription_id = str(parameters[self.PARAM_SUBSCRIBER_ID])
+    compute_client = ComputeManagementClient(credentials, subscription_id)
+    storage_account = parameters[self.PARAM_STORAGE_ACCOUNT]
+
+    # First we must deallocate the VM we wish to capture.
+    vm_deallocate = compute_client.virtual_machines.deallocate(resource_group,
+                                                               instance_id)
+    # Wait until we have finished deallocating.
+    vm_deallocate.wait()
+
+    # Then we generalize it.
+    compute_client.virtual_machines.generalize(resource_group,
+                                               instance_id)
+    # Safety precaution sleep.
+    time.sleep(self.SLEEP_TIME)
```

Comment:
Same comment as before, here too you could assign the generalize operation to a result object and check the done() method on it instead of waiting an arbitrary sleep time.